### PR TITLE
Tidy up the local js

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
- Removed jQuery from every page and added it to the form pages
- Removed import protocols
- Replace http iframes to https

### QA
Sorry this is again a long QA. I think I have now removed all YUI completely once this and the scratch branch lands. Apart from the sliders. They will require a discussion and a follow up branch.
 
 - [x] Go to /about/contact-us/form and check there are no js errors. Submit an empty form and it should validate
 - [x] Go to /cloud/contact-us and check there are no js errors. Submit an empty form and it should validate
 - [x] Go to /cloud and check there are no js errors. Check the D3 chart renders in the "Use Ubuntu on certified public clouds" section
 - [x] Go to /cloud/openstack and check there are no js errors.
 - [x] Go to /cloud/openstack/managed-cloud and check there are no js errors.
 - [x] Go to /cloud/openstack/managed-cloud/contact-us and check there are no js errors.
 - [x] Go to /cloud/openstack/managed-cloud/thank-you you will see a jQuery error but that is resolved in the Tidy up scratch branch
 - [x] Go to /cloud/partners/ You will find some YUI in the footer but that has been replaced in the Tidy up scratch branch
 - [x] Go to /cloud/storage/contact-us and you should find no js errors
 - [x] Go to /cloud/thank-you and you should see a missing jQuery dep. This function is coming from scratch and has been rewritten in ES5 in the Tidy up scratch branch
 - [x] Go to /cloud/training/classroom-contact-us and check there are no js errors
 - [x] Go to /cloud/training/contact-us and there should be no js errors
 - [x] Go to /cloud/training/onsite-contact-us and check there are no errors
 - [x] Go to /cloud/training/server-admin-onsite-contact-us and check there are no js errors
 - [x] Go to /cloud/training/thank-you and you will see a jQuery in scratch that is resolved with the Tidy up scratch branch
 - [x] Go to /desktop/contact-us and you should see no errors
 - [x] Go to /desktop/features and scroll to the 'Photos' section and it should have a background image with a caption in the bottom right. There will be a jQuery error from scratch but that is fixed in the scratch branch
 - [x] Go to /desktop/partners and scroll to 'A SELECTION OF OUR DESKTOP PARTNERS' and check there is a list of logos
 - [x] Go to /desktop/tenders-contact-us and check there are no js errors and the validation works
 - [x] Go to /download/desktop/thank-you?version=16.04.1&architecture=amd64 and check the download starts
 - [x] Go to /download/server/thank-you-linuxone and check there are no errors
 - [x] Go to /download/server/thank-you?version=16.04.1&architecture=amd64 and check the download kicks off. There will be a jQuery error.
 - [x] Go to /internet-of-things/contact-us and chec kthere is no errors.
 - [x] Go to /internet-of-things/digital-signage/contact-us and check there are no errors.
 - [x] Go to /internet-of-things/digital-signage/thank-you and check there is just the jquery scratch error
 - [x] Go to /legal/contributors/submit and check you get the same errors as live
 - [x] Go to /legal/terms-and-policies/contact-us and check you get the same errors as live
 - [x] Go to /management/contact-us and check there are no errors and the validation works
 - [x] Go to /server/contact-us and check there is no errors
 - [x] Go to /tablet/contact-us and check for errors

### CODE REVIEW

File: templates/internet-of-things/thank-you.html [hunk 1]
```diff
@@ -33,11 +33,11 @@
 var google_conversion_value = 0;
 /* ]]> */
 </script>
-<script type="text/javascript" src="http://www.googleadservices.com/pagead/conversion.js">
+<script type="text/javascript" src="//www.googleadservices.com/pagead/conversion.js">
```
Comment (nottrobin) [inline templates/internet-of-things/thank-you.html]: Could we explicitly use HTTPS urls? Protocolless URLs are now (arguably) an antipattern. https://twitter.com/paul_irish/status/588502455530311680. Credit to @barrymcgee.

File: templates/internet-of-things/thank-you.html [hunk 2]
```diff
@@ -33,11 +33,11 @@
...
 </script>
 <noscript>
 <div style="display:inline;">
-<img height="1" width="1" style="border-style:none;" alt="" src="http://www.googleadservices.com/pagead/conversion/1012391776/?value=0&amp;label=-mBBCIC5vwMQ4L7f4gM&amp;guid=ON&amp;script=0"/>
+<img height="1" width="1" style="border-style:none;" alt="" src="//www.googleadservices.com/pagead/conversion/1012391776/?value=0&amp;label=-mBBCIC5vwMQ4L7f4gM&amp;guid=ON&amp;script=0"/>
```
Comment (nottrobin) [inline templates/internet-of-things/thank-you.html]: Ditto

File: templates/management/thank-you.html [hunk 3]
```diff
@@ -22,11 +22,11 @@
 var google_conversion_value = 0;
 /* ]]> */
 </script>
-<script type="text/javascript" src="http://www.googleadservices.com/pagead/conversion.js">
+<script type="text/javascript" src="//www.googleadservices.com/pagead/conversion.js">
```
Comment (nottrobin) [inline templates/management/thank-you.html]: HTTPS please

File: templates/management/thank-you.html [hunk 4]
```diff
@@ -22,11 +22,11 @@
...
 </script>
 <noscript>
 <div style="display:inline;">
-<img height="1" width="1" style="border-style:none;" alt="" src="http://www.googleadservices.com/pagead/conversion/1012391776/?value=0&amp;label=utP4CMDOwQMQ4L7f4gM&amp;guid=ON&amp;script=0"/>
+<img height="1" width="1" style="border-style:none;" alt="" src="//www.googleadservices.com/pagead/conversion/1012391776/?value=0&amp;label=utP4CMDOwQMQ4L7f4gM&amp;guid=ON&amp;script=0"/>
```
Comment (nottrobin) [inline templates/management/thank-you.html]: HTTPS please

File: templates/tablet/thank-you.html [hunk 5]
```diff
@@ -36,11 +36,11 @@
 var google_conversion_value = 0;
 /* ]]> */
 </script>
-<script type="text/javascript" src="http://www.googleadservices.com/pagead/conversion.js">
+<script type="text/javascript" src="//www.googleadservices.com/pagead/conversion.js">
```
Comment (nottrobin) [inline templates/tablet/thank-you.html]: HTTPS please

File: templates/tablet/thank-you.html [hunk 6]
```diff
@@ -36,11 +36,11 @@
...
 </script>
 <noscript>
 <div style="display:inline;">
-<img height="1" width="1" style="border-style:none;" alt="" src="http://www.googleadservices.com/pagead/conversion/1012391776/?value=0&amp;label=-mBBCIC5vwMQ4L7f4gM&amp;guid=ON&amp;script=0"/>
+<img height="1" width="1" style="border-style:none;" alt="" src="//www.googleadservices.com/pagead/conversion/1012391776/?value=0&amp;label=-mBBCIC5vwMQ4L7f4gM&amp;guid=ON&amp;script=0"/>
```
Comment (nottrobin) [inline templates/tablet/thank-you.html]: HTTPS please

File: templates/desktop/tenders/contact-us.html [hunk 7]
```diff
@@ -1,146 +0,0 @@
-{% extends "desktop/base_desktop.html" %}
```
Comment (nottrobin) [inline templates/desktop/tenders/contact-us.html]: Isn't this page needed? It's linked to from <http://127.0.0.1:8001/desktop/partners> (although that page should be using a relative rather than absolute URL).

Comment (nottrobin) [inline templates/desktop/tenders/contact-us.html]: Oh no sorry I see, it redirects to `/desktop/tenders-contact-us.html`. Fair enough.

File: templates/about/contact-us/form/thank-you.html [hunk 8]
```diff
@@ -43,11 +43,11 @@
 var google_conversion_value = 0;
 /* ]]> */
 </script>
-<script type="text/javascript" src="http://www.googleadservices.com/pagead/conversion.js">
+<script type="text/javascript" src="//www.googleadservices.com/pagead/conversion.js">
```
Comment (nottrobin) [inline templates/about/contact-us/form/thank-you.html]: HTTPS please

File: templates/about/contact-us/form/thank-you.html [hunk 9]
```diff
@@ -43,11 +43,11 @@
...
 </script>
 <noscript>
 <div style="display:inline;">
-<img height="1" width="1" style="border-style:none;" alt="" src="http://www.googleadservices.com/pagead/conversion/1012391776/?value=0&amp;label=utP4CMDOwQMQ4L7f4gM&amp;guid=ON&amp;script=0"/>
+<img height="1" width="1" style="border-style:none;" alt="" src="//www.googleadservices.com/pagead/conversion/1012391776/?value=0&amp;label=utP4CMDOwQMQ4L7f4gM&amp;guid=ON&amp;script=0"/>
```
Comment (nottrobin) [inline templates/about/contact-us/form/thank-you.html]: HTTPS please

File: templates/cloud/thank-you.html [hunk 10]
```diff
@@ -9,7 +9,7 @@
 {% block content %}
 {% include "shared/_thank_you.html" with thanks_context="enquiring about Cloud" %}
 
-	
+
 {% endblock content %}
 {% block footer_extra %}
 <!-- Google Code for Services Canonical &mdash; contact us Conversion Page -->
```
Comment (nottrobin) [inline templates/cloud/thank-you.html]: I'm pretty sure this block of code for google ad services is exactly duplicated many times throughout the codebase. It could probably benefit from being an include.

File: templates/cloud/thank-you.html [hunk 11]
```diff
@@ -23,11 +23,11 @@
 var google_conversion_value = 0;
 /* ]]> */
 </script>
-<script type="text/javascript" src="http://www.googleadservices.com/pagead/conversion.js">
+<script type="text/javascript" src="//www.googleadservices.com/pagead/conversion.js">
```
Comment (nottrobin) [inline templates/cloud/thank-you.html]: HTTPS please

File: templates/cloud/thank-you.html [hunk 12]
```diff
@@ -23,11 +23,11 @@
...
 </script>
 <noscript>
 <div style="display:inline;">
-<img height="1" width="1" style="border-style:none;" alt="" src="http://www.googleadservices.com/pagead/conversion/1012391776/?value=0&amp;label=-mBBCIC5vwMQ4L7f4gM&amp;guid=ON&amp;script=0"/>
+<img height="1" width="1" style="border-style:none;" alt="" src="//www.googleadservices.com/pagead/conversion/1012391776/?value=0&amp;label=-mBBCIC5vwMQ4L7f4gM&amp;guid=ON&amp;script=0"/>
```
Comment (nottrobin) [inline templates/cloud/thank-you.html]: HTTPS please

File: templates/cloud/training/thank-you.html [hunk 13]
```diff
@@ -25,11 +25,11 @@
 var google_conversion_value = 0;
 /* ]]> */
 </script>
-<script type="text/javascript" src="http://www.googleadservices.com/pagead/conversion.js">
+<script type="text/javascript" src="//www.googleadservices.com/pagead/conversion.js">
```
Comment (nottrobin) [inline templates/cloud/training/thank-you.html]: HTTPS please

File: templates/cloud/training/thank-you.html [hunk 14]
```diff
@@ -25,11 +25,11 @@
...
 </script>
 <noscript>
 <div style="display:inline;">
-<img height="1" width="1" style="border-style:none;" alt="" src="http://www.googleadservices.com/pagead/conversion/1012391776/?value=0&amp;label=-mBBCIC5vwMQ4L7f4gM&amp;guid=ON&amp;script=0"/>
+<img height="1" width="1" style="border-style:none;" alt="" src="//www.googleadservices.com/pagead/conversion/1012391776/?value=0&amp;label=-mBBCIC5vwMQ4L7f4gM&amp;guid=ON&amp;script=0"/>
```
Comment (nottrobin) [inline templates/cloud/training/thank-you.html]: HTTPS please

File: templates/desktop/partners.html [hunk 15]
```diff
@@ -121,8 +121,10 @@ <h2 class="twelve-col">What our partners are saying</h2>
 {% endblock content %}
 {% block footer_extra %}
 <script>
-    partnersApiParams = '?programme__name=Desktop&featured=true';
-    partnersElementId = '#hardware-partners';
+  partnerLogoClouds = [{
```
Comment (nottrobin) [inline templates/desktop/partners.html]: As previously discussed, could we change this to being a function call?

File: templates/desktop/shared/_share-this.html [hunk 16]
```diff
@@ -1,5 +1,5 @@
 <div class="share-this">
-    <iframe src="http://www.facebook.com/plugins/like.php?href=http%3A%2F%2Fwww.ubuntu.com%2Fubuntu&amp;send=false&amp;layout=button_count&amp;show_faces=false&amp;action=like&amp;colorscheme=light&amp;font&amp;height=20" style="border:none; overflow:hidden; height:21px; padding-right:23px;"></iframe>
+    <iframe src="//www.facebook.com/plugins/like.php?href=http%3A%2F%2Fwww.ubuntu.com%2Fubuntu&amp;send=false&amp;layout=button_count&amp;show_faces=false&amp;action=like&amp;colorscheme=light&amp;font&amp;height=20" style="border:none; overflow:hidden; height:21px; padding-right:23px;"></iframe>
```
Comment (nottrobin) [inline templates/desktop/shared/_share-this.html]: HTTPS please

File: templates/download/server/linuxone-signup-thank-you.html [hunk 17]
```diff
@@ -37,11 +37,11 @@
 var google_conversion_value = 0;
 /* ]]> */
 </script>
-<script type="text/javascript" src="http://www.googleadservices.com/pagead/conversion.js">
+<script type="text/javascript" src="//www.googleadservices.com/pagead/conversion.js">
```
Comment (nottrobin) [inline templates/download/server/linuxone-signup-thank-you.html]: HTTPS please

File: templates/download/server/linuxone-signup-thank-you.html [hunk 18]
```diff
@@ -37,11 +37,11 @@
...
 </script>
 <noscript>
     <div style="display:inline;">
-        <img height="1" width="1" style="border-style:none;" alt="" src="http://www.googleadservices.com/pagead/conversion/1012391776/?value=0&amp;label=utP4CMDOwQMQ4L7f4gM&amp;guid=ON&amp;script=0"/>
+        <img height="1" width="1" style="border-style:none;" alt="" src="//www.googleadservices.com/pagead/conversion/1012391776/?value=0&amp;label=utP4CMDOwQMQ4L7f4gM&amp;guid=ON&amp;script=0"/>
```
Comment (nottrobin) [inline templates/download/server/linuxone-signup-thank-you.html]: HTTPS please

File: templates/internet-of-things/digital-signage/thank-you.html [hunk 19]
```diff
@@ -33,11 +33,11 @@
 var google_conversion_value = 0;
 /* ]]> */
 </script>
-<script type="text/javascript" src="http://www.googleadservices.com/pagead/conversion.js">
+<script type="text/javascript" src="//www.googleadservices.com/pagead/conversion.js">
```
Comment (nottrobin) [inline templates/internet-of-things/digital-signage/thank-you.html]: HTTPS please

File: templates/internet-of-things/digital-signage/thank-you.html [hunk 20]
```diff
@@ -33,11 +33,11 @@
...
 </script>
 <noscript>
 <div style="display:inline;">
-<img height="1" width="1" style="border-style:none;" alt="" src="http://www.googleadservices.com/pagead/conversion/1012391776/?value=0&amp;label=-mBBCIC5vwMQ4L7f4gM&amp;guid=ON&amp;script=0"/>
+<img height="1" width="1" style="border-style:none;" alt="" src="//www.googleadservices.com/pagead/conversion/1012391776/?value=0&amp;label=-mBBCIC5vwMQ4L7f4gM&amp;guid=ON&amp;script=0"/>
```
Comment (nottrobin) [inline templates/internet-of-things/digital-signage/thank-you.html]: HTTPS please

File: templates/internet-of-things/partners.html [hunk 21]
```diff
@@ -153,10 +153,9 @@ <h2 class="twelve-col">Commercial information</h2>
 {% endblock content %}
 {% block footer_extra %}
   <script>
-      YUI().use('event-base', function (Y) {
-          Y.on('domready', function () {
-              core.loadPartners('?programme__name=Internet%20of%20Things', '#iot-partners');
-          });
-      });
+      partnerLogoClouds = [{
```
Comment (nottrobin) [inline templates/internet-of-things/partners.html]: I'm guessing this ain't gonna work without `partners.js` being loaded afterwards.

In any case, let's move this to a function call instead.

File: templates/internet-of-things/thank-you.html (hunk 1 above)
Comment (anthonydillon) [inline templates/internet-of-things/thank-you.html]: Okie doke

**Comment by nottrobin** [comment]
Wanna do this as a driveby? https://github.com/ubuntudesign/www.ubuntu.com/issues/847

File: templates/cloud/thank-you.html (hunk 10 above)
Comment (anthonydillon) [inline templates/cloud/thank-you.html]: Yea spotted that. Alot of moving code in this branch. Will do a polish branch after.

**Comment by anthonydillon** [comment]
Thanks for the review @nottrobin. Rebased and explicitly using HTTPS urls throughout.

**Comment by anthonydillon** [comment]
TL; DR. Na, forms need a rethink.

I want to bring up the number of forms across the site and the different methods they have been implemented. I think as they seem to make up such a large proportion of the site need to develop a elegant and consistent method.

**Comment by nottrobin** [comment]
Lol how could anyone find your 1 paragraph too long?

**Comment by nottrobin** [comment]
@anthonydillon I filed your forms point as issue #851.

File: static/js/partners.js [hunk 22]
```diff
@@ -15,7 +15,7 @@ var partners = {}
 /**
  * Loops through the partner clouds and loads the feeds for each.
  */
-partners.loadPartnerClouds = function() {
+partners.loadPartnerClouds = function(partnerLogoClouds) {
```
Comment (nottrobin) [inline static/js/partners.js]: I would rather this was:

``` javascript
partners.loadPartnerCloud = function (queryParams, elementId) {
```

But if you want to merge this for now I can tidy it up afterwards.

File: static/js/partners.js [hunk 23]
```diff
@@ -93,8 +93,8 @@ partners.loadJSONP = function(uri, callback) {
 
 
 // Check partnerLogoClouds is available and if not produce warning
-if (typeof(partnerLogoClouds) === 'object') {
-    partners.loadPartnerClouds();
-} else {
-    console.warn('Load partner logos: Expected array "partnerLogoClouds" not found. Stopping.');
-}
+// if (typeof(partnerLogoClouds) === 'object') {
```
Comment (nottrobin) [inline static/js/partners.js]: Likewise, it would be nice not to keep commented out code, but I'm happy to tidy this up myself afterwards, since I need to do some work for the managed cloud calculator anyway,

**Comment by nottrobin** [comment]
@anthonydillon thanks for doing that. But `templates/server/index.html` still contains a `partnerLogoClouds` variable, which presumably won't work now.

**Comment by nottrobin** [comment]
<http://127.0.0.1:8001/cloud/training/onsite-contact-us> doesn't seem to be using the validate library. I just see HTML5 validation in Firefox. Is this what you meant by forms being inconsistent? Or is it unintentional?

**Comment by nottrobin** [comment]
All LGTM and works. Great work! Merging.
